test/integration/grpc_client_test.go

```diff
@@ -0,0 +1,169 @@
+
+	// Enable debug logging for instrumentation
+	t.Setenv("OTEL_LOG_LEVEL", "debug")
+
+	t.Log("Building instrumented gRPC client and server...")
+
+	// Build the server and client applications with the instrumentation tool
+	app.Build(t, serverDir, "go", "build", "-a")
+	app.Build(t, clientDir, "go", "build", "-a")
+
+	t.Log("Starting gRPC server...")
+
+	// Start the server
+	serverApp, outputPipe := app.Start(t, serverDir)
+	defer func() {
+		if serverApp.Process != nil {
+			_ = serverApp.Process.Kill()
+		}
+	}()
+
+	readyChan := make(chan struct{})
+	go func() {
+		serverOutput := strings.Builder{}
+		scanner := bufio.NewScanner(outputPipe)
+		for scanner.Scan() {
+			line := scanner.Text()
+			serverOutput.WriteString(line + "\n")
+			if strings.Contains(line, "server started") {
+				close(readyChan)
+			}
+		}
+	}()
+
+	// Wait for server to be ready
+	select {
+	case <-readyChan:
+		t.Log("Server is ready!")
+	case <-time.After(5 * time.Second):
+		t.Fatal("timeout waiting for server to be ready")
+	}
+
+	t.Log("Running gRPC client (unary RPC)...")
+
+	// Test unary RPC - this uses grpc.NewClient internally
+	clientOutput := app.Run(t, clientDir, "-name", "ClientTest")
+	require.Contains(t, clientOutput, `"msg":"greeting"`, "Expected greeting response from server")
+	require.Contains(t, clientOutput, `"message":"Hello ClientTest"`, "Expected personalized greeting")
+
+	t.Log("Verifying client instrumentation...")
+
+	// Verify client instrumentation hooks were called
+	require.Contains(
+		t,
+		clientOutput,
+		"gRPC client instrumentation initialized",
+		"client instrumentation should be initialized",
+	)
+	require.Contains(t, clientOutput, "BeforeNewClient called", "client before hook should be called")
+	require.Contains(t, clientOutput, "AfterNewClient called", "client after hook should be called")
+
+	t.Log("Shutting down server...")
+
+	// Send shutdown
+	shutdownOutput := app.Run(t, clientDir, "-shutdown")
+	require.Contains(t, shutdownOutput, `"msg":"shutdown response"`, "Expected shutdown response")
+
+	// Wait for server to exit
+	_ = serverApp.Wait()
+
+	t.Log("gRPC client integration test passed!")
+}
+
+// TestGRPCClientServerStreaming tests gRPC streaming RPC instrumentation
+func TestGRPCClientServerStreaming(t *testing.T) {
+	serverDir := filepath.Join("..", "..", "demo", "grpc", "server")
+	clientDir := filepath.Join("..", "..", "demo", "grpc", "client")
+
+	// Enable debug logging for instrumentation
+	t.Setenv("OTEL_LOG_LEVEL", "debug")
+
+	t.Log("Building instrumented gRPC applications...")
+
+	// Build the applications
+	app.Build(t, serverDir, "go", "build", "-a")
+	app.Build(t, clientDir, "go", "build", "-a")
+
+	t.Log("Starting gRPC server...")
+
+	// Start the server
+	serverApp, outputPipe := app.Start(t, serverDir)
+	defer func() {
+		if serverApp.Process != nil {
+			_ = serverApp.Process.Kill()
+		}
+	}()
+
+	readyChan := make(chan struct{})
+	go func() {
+		serverOutput := strings.Builder{}
```

Comment:
This is probably also related to `http`, but I would suggest to move this line above, to get better error message